salt/utils/cloud.py

```diff
@@ -88,6 +88,10 @@
 except ImportError:
     HAS_GETPASS = False
 
+# This is required to support international characters in AWS EC2 tags or any
+# other kind of metadata provided by particular Cloud vendor.
+MSGPACK_ENCODING = 'utf-8'
```

Comment:
Can we just use `__salt_system_encoding__` instead?

Reply:
@gtmanfred I'm not sure, really.

In my understanding, the locale of the system that Salt is running on has nothing to do with any weird characters, which could potentially appear in a Cloud instance metadata. They could be filled in manually, or by another automation system. Anyhow, Salt Cloud should be able to write everything on disk and restore without loosing bits.
I think `utf-8` is best choice here and it seems that `msgpack` supports it well by default.

But if you think that it should be defined with `__salt_system_encoding__`, we probably need to update the docs that host (or user) running `salt-cloud` should have proper locale set up.

Reply:
In contrast, if any unicode symbol somehow appears in some instance metadata, and you would not set the correct locale on the Salt host, the `salt-cloud` would entirely fail with `UnicodeDecodeError` exception.
That's the case when it's really hard to figure out what you should do with that.

@gtmanfred Please just let me know what do think would be the better way to handle this?

Reply:
So, the problem is that if the systems local is not setup to handle utf-8, we shouldn't handle it.

Reply:
Okay. That makes sense. I will follow up with documentation update for Salt Cloud to cover necessary locale setup.

Thanks.

Reply:
utf-8 is correct as this is what is coming from AWS.    ``__salt_system_encoding__``  is wrong as it could be anything and does not represent the results of an AWS REST API.  The data must be stored in a Unicode compatible format so it can be store in variables within Salt as Unicode.

The salt cli is responsible to convert the output of any salt-cloud command to __salt_system_encoding__ from internal Unicode variables. (I suspect PY2 & PY3 already do this in the log library)

@terminalmage will be adding code to convert cli arguments to Unicode (PY2 only)